internal/gatewayapi/helpers.go

```diff
@@ -197,3 +197,13 @@ func hostnameMatchesWildcardHostname(hostname, wildcardHostname string) bool {
 	wildcardMatch := strings.TrimSuffix(hostname, strings.TrimPrefix(wildcardHostname, "*"))
 	return len(wildcardMatch) > 0
 }
+
+// containsPortNum tells whether a contains b.
+func containsPortNum(a []v1beta1.PortNumber, b v1beta1.PortNumber) bool {
+	for _, n := range a {
+		if b == n {
+			return true
+		}
+	}
+	return false
+}
```

Comment:
Pre-Go-1.18, this would have been a necessary function. But now we have [`golang.org/x/exp/slices.Contains`](https://pkg.go.dev/golang.org/x/exp/slices#Contains).